rust/stackable-cockpit/src/platform/release/spec.rs

```diff
@@ -108,11 +121,85 @@ impl ReleaseSpec {
             .await
     }
 
+    /// Upgrades a release by upgrading individual operators.
+    #[instrument(skip_all, fields(
+        %namespace,
+    ))]
+    pub async fn upgrade_crds(
+        &self,
+        include_products: &[String],
+        exclude_products: &[String],
+        namespace: &str,
+        k8s_client: &Client,
+    ) -> Result<()> {
+        info!("Upgrading CRDs for release");
+
+        include_products.iter().for_each(|product| {
+            Span::current().record("product.included", product);
+        });
+        exclude_products.iter().for_each(|product| {
+            Span::current().record("product.excluded", product);
+        });
+
+        let client = reqwest::Client::new();
+        let operators = self.filter_products(include_products, exclude_products);
+
+        for (product_name, product) in operators {
+            info!("Upgrading CRDs for {product_name}-operator");
+
+            let release_branch = match product.version.pre.as_str() {
+                "dev" => "main".to_string(),
+                _ => {
+                    format!("{}", product.version)
+                }
+            };
+
+            let request_url = format!(
+                "https://raw.githubusercontent.com/stackabletech/{product_name}-operator/{release_branch}/deploy/helm/{product_name}-operator/crds/crds.yaml"
+            );
+
+            // Get CRD manifests from request_url
+            let response = client
```

Comment:
I think we might want to use the transfer client which supports caching (just so repeated calls in a short timeframe don't make more requests).

Reply:
I'm not super familiar with TransferClient itself, so maybe it doesn't make sense here.